src/components/activity/ActivityOrganizerCollectionBlock.tsx

```diff
@@ -0,0 +1,117 @@
+  font-weight: bold;
+`
+
+const UpdateActivityOrganizer = gql`
+  mutation UpdateActivityOrganizer($activityId: uuid!, $organizerId: String!, $updatedAt: timestamptz!) {
+    update_activity(where: { id: { _eq: $activityId } }, _set: { organizer_id: $organizerId, updated_at: $updatedAt }) {
+      affected_rows
+    }
+  }
+`
+
+const ActivityOrganizerCollectionBlock: React.FC<{
+  activity: ActivityProps | null
+  onRefetch?: () => void
+}> = ({ activity, onRefetch }) => {
+  const { formatMessage } = useIntl()
+  const { member } = usePublicMember(activity?.organizerId || '')
+  const [updateActivityOrganizer] = useMutation<
+    hasura.UpdateActivityOrganizer,
+    hasura.UpdateActivityOrganizerVariables
+  >(UpdateActivityOrganizer)
+
+  const [visible, setVisible] = useState(false)
+  const [loading, setLoading] = useState(false)
+  const [selectedMemberId, setSelectedMemberId] = useState<string | null>(null)
+
+  if (!activity) {
+    return <Skeleton active />
+  }
+
+  const handleSubmit = () => {
+    if (!selectedMemberId) {
+      return
+    }
+
+    setLoading(true)
+
+    updateActivityOrganizer({
+      variables: {
+        activityId: activity.id,
+        organizerId: selectedMemberId,
+        updatedAt: new Date(),
+      },
+    })
+      .then(() => {
+        message.success(formatMessage(commonMessages.event.successfullySaved))
+        setVisible(false)
+        setSelectedMemberId(null)
+        onRefetch?.()
+      })
+      .catch(handleError)
+      .finally(() => setLoading(false))
+  }
+
+  return (
+    <>
+      <RoleAdminBlock name={member.name} pictureUrl={member.pictureUrl} onEdit={() => setVisible(true)} />
+
+      <Modal
+        title={null}
+        footer={null}
+        centered
+        destroyOnClose
+        visible={visible}
+        onCancel={() => {
+          setVisible(false)
+          setSelectedMemberId('')
+        }}
+      >
+        <StyledModalTitle className="mb-4">{formatMessage(commonMessages.ui.edit)}</StyledModalTitle>
+
+        <Form layout="vertical" colon={false} hideRequiredMark onFinish={handleSubmit}>
+          <Form.Item label={formatMessage(commonMessages.label.selectOrganizer)}>
+            <ContentCreatorSelector
+              value={selectedMemberId || member.id}
+              onChange={value => setSelectedMemberId(value)}
+              allowedPermissions={['ACTIVITY_NORMAL']}
+            />
+          </Form.Item>
+
+          <Form.Item className="text-right">
+            <Button
+              onClick={() => {
+                setVisible(false)
+                setSelectedMemberId('')
+              }}
+              className="mr-2"
+            >
+              {formatMessage(commonMessages.ui.cancel)}
+            </Button>
+            <Button type="primary" htmlType="submit" loading={loading}>
+              {formatMessage(commonMessages.ui.save)}
+            </Button>
+          </Form.Item>
+        </Form>
+      </Modal>
+    </>
+  )
```

Comment:
The creator's block is missing

Reply:
The creator of the column in the current data table is the same as the organizer, which is why I didn't add a creator block.
Do you think it would be better to add a new column for the creator, or do you have any other ideas?

Reply:
But the requirement indicates that both should be implemented, so I think it should be added. 
If there are any confusion about the requirement, you should ask ZZ to discuss.

![image](https://github.com/urfit-tech/lodestar-app-admin/assets/64723534/5ee21530-25f7-4b53-a1fe-0ad9163e258c)

Reply:
> Do you think it would be better to add a new column for the creator, or do you have any other ideas?
I think we can refer to the approach of `program` and add table `activity_role` to  complete this requirement.

Reply:
I have confirmed the requirements with ZZ. Currently, there is no need to create a creator section. Can discuss the details when this requirement arises next time. The discussion results have been documented in the card